plugins/cairo-lang-macro-attributes/src/lib.rs

```diff
@@ -28,3 +28,40 @@ pub fn attribute_macro(_args: TokenStream, input: TokenStream) -> TokenStream {
     };
     TokenStream::from(expanded)
 }
+
+/// This macro can be used to construct the auxiliary data collection callback.
+///
+/// The procedural macro can emit additional auxiliary data alongside the generated [`TokenStream`]
+/// during the code expansion. This data can be used to collect additional information from the
+/// source code of a project that is being compiled during the macro execution.
+/// For instance, you can create a procedural macro that collects some information stored by
+/// the Cairo programmer as attributes in the project source code.
+///
+/// This should be used to implement a collection callback for the auxiliary data.
+/// This callback will be called after the source code compilation (and thus after all the procedural
+/// macro executions). All auxiliary data emitted by the procedural macro during source code compilation
+/// will be passed to the callback as an argument.
+///
+/// The callback can be used to process or persist the data collected during the compilation.
+///
+/// This macro hides the conversion to stable ABI structs from the user.
+///
+/// # Safety
+/// Note that AuxData deserialization may fail.
+#[proc_macro_attribute]
+pub fn aux_data_collection_callback(_args: TokenStream, input: TokenStream) -> TokenStream {
+    let mut item: ItemFn = parse_macro_input!(input as ItemFn);
+    // Rename item to hide it from the macro source code.
+    let id = uuid::Uuid::new_v4().to_string().replace('-', "");
```

Comment:
This is non-deterministic and may bust some caches 

a better approach would be to hash the identifier using stable hasher like one we use in Scarb proper